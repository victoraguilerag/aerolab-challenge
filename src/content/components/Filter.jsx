import React from 'react'
import PropTypes from 'prop-types'

function Filter (props) {
  const { handleFilterChange, active } = props
  if (active) {
    return (
      <li className="active-filter" style={styles.activeFilter} onClick={handleFilterChange}>
        <p className="active-filter-text" style={styles.activeFilterText}>
          {props.label}
        </p>
      </li>
    )
  }
  return (
    <li className="filter" style={styles.filter} onClick={handleFilterChange}>
      <p className="filter-text" style={styles.filterText}>
        {props.label}
      </p>
    </li>
  )
}

Filter.propTypes = {
  handleFilterChange: PropTypes.func.isRequired,
  active: PropTypes.bool.isRequired,
  label: PropTypes.string.isRequired
}

const styles = {
  'filter': {
    'userSelect': 'none',
    'background': '#ededed',
    'borderRadius': '100px',
    'width': '163px',
    'height': '48px',
    'margin': '0 5px',
    'marginLeft': '2%',
    'cursor': 'pointer'
  },
  'filterText': {
    'userSelect': 'none',
    'fontSize': '1em',
    'color': '#a3a3a3',
    'height': '100%',
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'textAlign': 'center',
    'margin': '0',
    'padding': '0 10px'
  },
  'activeFilter': {
    'background': '#0ad4fa',
    'borderRadius': '100px',
    'width': '163px',
    'height': '48px',
    'margin': '0 5px',
    'marginLeft': '2%',
    'cursor': 'pointer'
  },
  'activeFilterText': {
    'userSelect': 'none',
    'color': '#ffffff',
    'fontSize': '1em',
    'height': '100%',
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'textAlign': 'center',
    'margin': '0',
    'padding': '0 10px'
  }
}

export default Filter
