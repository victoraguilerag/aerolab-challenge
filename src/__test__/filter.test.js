import React from 'react'
import Filter from '../content/components/Filter.jsx'
import renderer from 'react-test-renderer'

test('Filter renders', () => {
  const component = renderer.create(
    <Filter
      handleFilterChange={()=>console.log('handle filter change')}
      active={true}
      label="Test label"
    />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
