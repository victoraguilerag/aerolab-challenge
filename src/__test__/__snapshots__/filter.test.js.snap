// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders 1`] = `
<li
  className="active-filter"
  onClick={[Function]}
  style={
    Object {
      "background": "#0ad4fa",
      "borderRadius": "100px",
      "cursor": "pointer",
      "height": "48px",
      "margin": "0 5px",
      "marginLeft": "2%",
      "width": "163px",
    }
  }
>
  <p
    className="active-filter-text"
    style={
      Object {
        "alignItems": "center",
        "color": "#ffffff",
        "display": "flex",
        "fontSize": "1em",
        "height": "100%",
        "justifyContent": "center",
        "margin": "0",
        "padding": "0 10px",
        "textAlign": "center",
        "userSelect": "none",
      }
    }
  >
    Test label
  </p>
</li>
`;
